wagtail_newsletter/campaign_backends/__init__.py

```diff
@@ -0,0 +1,47 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+
+DEFAULT_CAMPAIGN_BACKEND = (
+    "wagtail_newsletter.campaign_backends.mailchimp.MailchimpCampaignBackend"
+)
+
+
+@dataclass
+class Audience:
+    id: str
+    name: str
+    member_count: int
+
+
+@dataclass
+class AudienceSegment:
+    id: str
+    name: str
+    member_count: int
+
+
+class AudienceNotFoundError(LookupError):
+    pass
+
+
+class CampaignBackend:
```

Comment:
The class could be a better "abstract" one if it was a subclass of `ABC`?  

Without it one can still create an instance of a class that doesn't implement the contract, and it's only when its methods will be used that the errors will happen:  
  
![Screenshot from 2024-05-15 09-55-03](https://github.com/wagtail/wagtail-newsletter/assets/722388/d871e14e-f786-4b3c-b983-c031138b953a)

...versus trying to do that when the class uses `ABC`:  
  
![Screenshot from 2024-05-15 09-57-35](https://github.com/wagtail/wagtail-newsletter/assets/722388/3fb912ee-01b6-4615-91f7-b739df5e737a)

...which gives us a clear error message, and makes more sense to me from a business logic point of view, as the class is invalid if it doesn't implement the contract, and therefore shouldn't be "allowed" to be instantiated? :slightly_smiling_face:

Reply:
I think you're right. I hesitated to make it an actual ABC, because it makes it more difficult to create a mock subclass for testing, but that's not a great reason TBH. I'll make it a proper ABC.